docs/components/index.md

```diff
@@ -0,0 +1,351 @@
+
+In the following example, the `ParentComponent` sets the value of the `Title` property of the `ChildComponent`:
+
+*ParentComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ParentComponent.cshtml?start=1&end=7&highlight=5)]
+
+*ChildComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ChildComponent.cshtml?highlight=7)]
+
+## Child content
+
+Components can set the content of an element in another component by assigning to a property that represents a `RenderFragment` type.
+
+In the following example, the `ChildContent` element of the `ParentComponent` sets the value of the `ChildComponent`'s `PanelContent` property. `PanelContent` is a `RenderFragment` type and supplies the content to the Bootstrap panel.
+
+*ParentComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ParentComponent.cshtml?start=1&end=7&highlight=6)]
+
+*ChildComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ChildComponent.cshtml?highlight=3,8)]
+
+## Data binding
+
+Data binding to both components and DOM elements is accomplished with the `bind` attribute. The following example binds the `ItalicsCheck` property to the check box's checked state:
+
+```cshtml
+<input type="checkbox" class="form-check-input" id="italicsCheck" bind="@_italicsCheck" />
+```
+
+When the check box is selected and cleared, the property's value is updated to `true` and `false`, respectively.
+
+The check box is updated in the UI only when the component is rendered, not in response to changing the property's value. Since components render themselves after event handler code executes, property updates are usually reflected in the UI immediately.
+
+Using `bind` with a `CurrentValue` property (`<input bind="@CurrentValue" />`) is essentially equivalent to the following:
+
+```cshtml
+<input value="@CurrentValue" 
+    onchange="@((UIValueEventArgs __e) => CurrentValue = __e.Value) />
+```
+
+When the component is rendered, the `value` of the input element comes from the `CurrentValue` property. When the user types in the text box, the `onchange` is fired and the `CurrentValue` property is set to the changed value. In reality, the code generation is a little more complex because `bind` deals with a few cases where type conversions are performed. In principle, `bind` associates the current value of an expression with a `value` attribute and handles changes using the registered handler.
+
+**Format strings**
+
+Data binding works with [DateTime](https://docs.microsoft.com/dotnet/api/system.datetime) format strings (but not other format expressions at this time, such as currency or number formats):
+
+```cshtml
+<input bind="@StartDate" format-value="yyyy-MM-dd" />
+
+@functions {
+    public DateTime StartDate { get; set; } = new DateTime(2020, 1, 1);
+}
+```
+
+**Conditional attributes**
```

Comment:
This should really go in the "Razor support" section as it doesn't really have anything to do with data binding. It's just a Razor feature that we support.